crates/task-impls/src/consensus/mod.rs

```diff
@@ -249,6 +249,9 @@ impl<TYPES: NodeType, I: NodeImplementation<TYPES>> ConsensusTaskState<TYPES, I>
         let Some(proposal) = self.current_proposal.clone() else {
             return;
         };
+        if proposal.get_view_number() != view {
+            return;
+        }
```

Comment:
Since I'm a bit unfamiliar with the changes to the task system since I last looked at it:
Does returning out of here just throw out the proposal?

Reply:
Ah no, we just have a clone of it here, we only throw out a proposal after we vote (see `QuorumVoteSend` handler)